contributor-docs/code-contributions.md

```diff
@@ -0,0 +1,427 @@
+|-----------------------|----------------|-------------------------------------------------|-----------------|--------------------------------------------------------------------------------------------------------|
+| Datastore Bulk Delete | String         | A JSON string of the entity                     | String          | A JSON string of the entity to delete; filter entities by returning undefined                          |
+| Datastore to Pub/Sub  | String         | A JSON string of the entity                     | String          | The payload to publish to Pub/Sub                                                                      |
+| Datastore to GCS Text | String         | A JSON string of the entity                     | String          | A single-line within the output file                                                                   |
+| GCS Text to BigQuery  | String         | A single-line within the input file             | String          | A JSON string which matches the destination table's schema                                             |
+| Pub/Sub to BigQuery   | String         | A string representation of the incoming payload | String          | A JSON string which matches the destination table's schema                                             |
+| Pub/Sub to Datastore  | String         | A string representation of the incoming payload | String          | A JSON string of the entity to write to Datastore                                                      |
+| Pub/Sub to Splunk     | String         | A string representation of the incoming payload | String          | The event data to be sent to Splunk HEC events endpoint. Must be a string or a stringified JSON object |
+
+### UDF Examples
+
+For a comprehensive list of samples, please check our [udf-samples](v2/common/src/main/resources/udf-samples) folder.
+
+#### Adding fields
+
+```js
+/**
+ * A transform which adds a field to the incoming data.
+ * @param {string} inJson
+ * @return {string} outJson
+ */
+function transform(inJson) {
+  var obj = JSON.parse(inJson);
+  obj.dataFeed = "Real-time Transactions";
+  obj.dataSource = "POS";
+  return JSON.stringify(obj);
+}
+```
+
+#### Filtering records
+
+```js
+/**
+ * A transform function which only accepts 42 as the answer to life.
+ * @param {string} inJson
+ * @return {string} outJson
+ */
+function transform(inJson) {
+  var obj = JSON.parse(inJson);
+  // only output objects which have an answer to life of 42.
+  if (obj.hasOwnProperty('answerToLife') && obj.answerToLife === 42) {
+    return JSON.stringify(obj);
+  }
+}
+```
+
+## Generated Documentation
+
+This repository contains generated documentation, which contains a list of parameters
+and instructions on how to customize and/or build every template.
+
+To generate the documentation for all templates, the following command can be used:
+
+```shell
+mvn clean prepare-package \
+  -DskipTests \
+  -PtemplatesSpec
+```
+
```

Comment:
can we add the links for python flex template builds?
https://cloud.google.com/dataflow/docs/guides/templates/using-flex-templates#python_1
https://github.com/GoogleCloudPlatform/python-docs-samples/tree/main/dataflow/flex-templates